packages/radix-ui-themes/src/components/data-list.tsx

```diff
@@ -0,0 +1,72 @@
+import classNames from 'classnames';
+import * as React from 'react';
+import { MarginProps, GetPropDefTypes, extractProps, marginPropDefs } from '../helpers';
+import { dataListPropDefs, dataListItemPropDefs, dataListLabelPropDefs } from './data-list.props';
+
+type DataListRootOwnProps = GetPropDefTypes<typeof dataListPropDefs>;
+interface DataListRootProps
+  extends React.ComponentPropsWithoutRef<'dl'>,
+    MarginProps,
+    DataListRootOwnProps {}
+const DataListRoot = React.forwardRef<HTMLDListElement, DataListRootProps>(
+  (props, forwardedRef) => {
+    const { className, ...dataListProps } = extractProps(props, dataListPropDefs, marginPropDefs);
+    return (
+      <dl
+        ref={forwardedRef}
+        className={classNames(className, 'rt-DataListRoot', 'rt-Text')}
+        {...dataListProps}
+      />
+    );
+  }
+);
+DataListRoot.displayName = 'DataListRootGrid';
+
+interface DataListItemProps
+  extends React.ComponentPropsWithRef<'div'>,
```

Comment:
These should always use `ComponentPropsWithoutRef` when using `React.forwardRef` because then the typed `forwardRef` is the one adding the `ref` type./